### PR TITLE
sdk/js: Add instruction decoder to solana

### PR DESCRIPTION
**Pull request by guibescos**
The current sdk/js for solana is a little bit lackluster :
- There is no encoder for PostMessage
- There is no intruction decoder

I need both things for cross-chain governance tools. I decided to copy what anchor does for encoding/decoding.
I updated WormholeInstructionCoder to parse the layouts of the instructions from wormhole.json (wormhole's IDL). I then use these layouts in WormholeInstructionCoder to encode/decode instructions. This means we no longer need to hand-serialize/deserialize instructions. To add a new instructions you just need to add it to wormhole.json.

### CODE REVIEW

**Review by a5-pickle — DISMISSED** [review]
These changes are awesome 💪

File: sdk/js/src/solana/wormhole/coder/instruction.ts [hunk 1]
```diff
@@ -1,41 +1,88 @@
-import { Idl, InstructionCoder } from "@project-serum/anchor";
-import { ETHEREUM_KEY_LENGTH } from "../../utils";
+import { Idl, Instruction, InstructionCoder } from "@project-serum/anchor";
+import { bs58 } from "@project-serum/anchor/dist/cjs/utils/bytes";
```
Comment (a5-pickle) [inline sdk/js/src/solana/wormhole/coder/instruction.ts:2]: I didn't catch this the first time I reviewed it. Is this going to cause any issues with esm?

File: sdk/js/src/solana/wormhole/coder/idl.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
+// Borrowed from coral-xyz/anchor
```
Comment (a5-pickle) [inline sdk/js/src/solana/wormhole/coder/idl.ts:1]: Just making a comment here as a note that we would probably like this for Token Bridge and NFT Bridge program decoding.

**Review by a5-pickle — CHANGES_REQUESTED** [review]

File: sdk/js/src/solana/wormhole/coder/instruction.ts (hunk 1 above)
Comment (guibescos) [reply sdk/js/src/solana/wormhole/coder/instruction.ts]: Ok, so I just solved this by explicitly naming bs58 in `package.json`. bs58 was already in package-lock.json so I don't think this increases the size of the package.

**Review by a5-pickle — APPROVED** [review]

**Review by kev1n-peters — APPROVED** [review]

**Pull request merged**